### PR TITLE
Token2022 Support

### PR DESCRIPTION
**Pull request by losman0s**
### Highlights
* Adds support to Token2022 interface
* Account for transfer fees in internal accounting

### Notes
* Extension support is currently **not** extended to tokens with program hooks that require extra accounts. 
* For UX reasons, the logic to account for transfer fees in internal accounting does so by always having the recipient receive the amount requested for relevant actions, i.e.:
  * for **deposit** & **repay** the amount withdrawn from the user wallet will be inflated so that the amount deposited to the vault matches the amount requested.
  * for **withdraw** & **borrow**, the amount withdrawn from the vault is inflated so that the amount deposited to the user wallet matches the amount requested.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,28 +1,33 @@
 [workspace]
 resolver = "2"
-members = ["programs/*", "clients/rust/*", "tools/*", "observability/indexer"]
+members = [
+    "programs/*",
+    "clients/rust/*",
+    "tools/*",
+    #"observability/indexer"
```
[the remark this answers is not in the document]
Comment (losman0s) [reply Cargo.toml]: Removed

File: programs/marginfi/src/utils.rs [hunk 2]
```diff
@@ -27,3 +40,77 @@ where
...
+        transfer_fee_config
+            .calculate_inverse_epoch_fee(epoch, post_fee_amount)
+            .unwrap()
+    } else {
+        0
+    };
+
+    let pre_fee_amount = post_fee_amount
+        .checked_add(fee)
+        .ok_or(MarginfiError::MathError)?;
```
[the remark this answers is not in the document]
Comment (losman0s) [reply programs/marginfi/src/utils.rs]: Just saw this comment, was everything clarified there?

File: .github/actions/build-program/action.yaml
[the remark this answers is not in the document]
Comment (losman0s) [reply .github/actions/build-program/action.yaml:1]: Yes, earlier I was hitting some issues due to caching, and not looking to go down that rabbit hole now. Just makes things a bit slower.

File: .github/actions/build-verifiable-program/action.yaml
[the remark this answers is not in the document]
Comment (losman0s) [reply .github/actions/build-verifiable-program/action.yaml:1]: Ditto

File: .github/actions/build-workspace/action.yaml
[the remark this answers is not in the document]
Comment (losman0s) [reply .github/actions/build-workspace/action.yaml:1]: Ditto

File: .github/actions/setup-anchor-cli/action.yaml
[the remark this answers is not in the document]
Comment (losman0s) [reply .github/actions/setup-anchor-cli/action.yaml:1]: Ditto

File: .github/actions/setup-solana-cli/action.yaml [hunk 3]
```diff
@@ -3,20 +3,23 @@ description: "Setup Solana CLI"
...
-      - run: solana config set --url localhost
-        shell: bash
+    # - uses: actions/cache@v2
+    #   name: Cache Solana Tool Suite
+    #   id: cache-solana
+    #   with:
+    #     path: |
+    #       ~/.cache/solana/
+    #       ~/.local/share/solana/
+    #     key: solana-${{ runner.os }}-v0000-${{ env.SOLANA_CLI_VERSION }}-${{ env.RUST_TOOLCHAIN }}
```
[the remark this answers is not in the document]
Comment (losman0s) [reply .github/actions/setup-solana-cli/action.yaml:13]: Ditto

File: clients/rust/marginfi-cli/src/processor/mod.rs [hunk 4]
```diff
@@ -2315,18 +2367,18 @@ fn timestamp_to_string(timestamp: i64) -> String {
...
+//         .rpc()
+//         .get_account_data(aggregator_pk)
+//         .expect("Aggregator account not found");
+
+//     let aggregator_account =
+//         switchboard_solana::AggregatorAccountData::new_from_bytes(&aggregator_account_data)
+//             .expect("Invalid aggregator account data");
+
+//     todo!("print")
+// }
```
[the remark this answers is not in the document]
Comment (losman0s) [reply clients/rust/marginfi-cli/src/processor/mod.rs]: Removed

File: clients/rust/marginfi-cli/src/entrypoint.rs [hunk 5]
```diff
@@ -416,25 +400,24 @@ pub fn entry(opts: Opts) -> Result<()> {
...
+        //     let config = profile.get_config(Some(&opts.cfg_override))?;
 
-            processor::process_inspect_switchboard_feed(&config, &switchboard_feed);
+        //     processor::process_inspect_switchboard_feed(&config, &switchboard_feed);
 
-            Ok(())
-        }
-        #[cfg(feature = "dev")]
+        //     Ok(())
+        // }
```
[the remark this answers is not in the document]
Comment (losman0s) [reply clients/rust/marginfi-cli/src/entrypoint.rs]: Removed. Smth broke with the new crate and we can whip up smth new when needed.

**Review by jkbpvsc — APPROVED** [review]

**Pull request merged**